pkg/querier/blocks_store_replicated_set.go

```diff
@@ -135,25 +145,50 @@ func (s *blocksStoreReplicationSet) GetClientsFor(userID string, blockIDs []ulid
 			return nil, errors.Wrapf(err, "failed to get store-gateway client for %s", addr)
 		}
 
-		clients[c.(BlocksStoreClient)] = blockIDs
+		sgc, ok := c.(*storeGatewayClient)
+		if !s.zoneAwarenessEnabled || !ok {
+			clients[c.(BlocksStoreClient)] = blockIDs
+		} else {
+			bsc := &zoneAwareStoreGatewayClient{storeGatewayClient: sgc, zone: zones[addr]}
```

Comment:
Do we need a new type here? Can the zone be added to the existing client?

Reply:
Changes reverted so should be fixed.